src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImp;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImp;
+import core.basesyntax.exception.DaoAnnotationNotFound;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, Factory.getBetDao());
+                }
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, Factory.getUserDao());
+                }
```

Comment:
let's refactor lines 22-27 (lines 20-21 are ok):

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception that the field has `@Inject` annotation but there is no suitable implementation